salt/utils/files.py

```diff
@@ -616,6 +616,31 @@ def safe_filepath(file_path_name, dir_sep=None):
     return path
 
 
+def is_encoding(path, encoding="utf-8"):
+    '''
+    Detect if the file can be successfully decoded by the passed encoding
+
+    Args:
+
+        fp_ (pointer): A pointer to the file to check
+        encoding (str): The encoding to test
+
+    Return:
+        bool: True if successful, otherwise False
+    '''
+    if not os.path.isfile(path):
+        return False
```

Comment:
I feel like we should probably be attempting to read from the file and simply raising the exception here, but it looks like `is_binary`, et al have already established the pattern that we just return `False` in these cases, so I guess this is OK.